Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs

```diff
@@ -62,46 +76,63 @@ private void Start()
                 // If we don't have what we need to proceed, we may as well remove ourselves.
                 Destroy(this);
             }
+
+            if (PlaceParentOnTap)
+            {
+                if  (ParentGameObjectToPlace != null && !gameObject.transform.IsChildOf(ParentGameObjectToPlace.transform))
+                {
+                    Debug.LogError("The specified parent object is not a parent of this object.");
+                }
+
+                DetermineParent();
+            }
         }
 
-        private void Update()
+        public virtual void Update()
         {
             // If the user is in placing mode,
             // update the placement to match the user's gaze.
-            if (placing)
+            if (IsBeingPlaced)
             {
+                if (gameObject.GetComponent<TapToPlace>().IsBeingPlaced) { }
                 // Do a raycast into the world that will only hit the Spatial Mapping mesh.
-                var headPosition = Camera.main.transform.position;
-                var gazeDirection = Camera.main.transform.forward;
+                Vector3 headPosition = Camera.main.transform.position;
+                Vector3 gazeDirection = Camera.main.transform.forward;
 
                 RaycastHit hitInfo;
-                if (Physics.Raycast(headPosition, gazeDirection, out hitInfo,
-                    30.0f, spatialMappingManager.LayerMask))
+                if (Physics.Raycast(headPosition, gazeDirection, out hitInfo, 30.0f, spatialMappingManager.LayerMask))
                 {
+                    // Place the parent object as well but keep the focus on the current game object
+                    if (PlaceParentOnTap)
+                    {
+                        Vector3 relativePositionToParent = gameObject.transform.position - ParentGameObjectToPlace.transform.position;
+                        ParentGameObjectToPlace.transform.position = hitInfo.point - relativePositionToParent;
+                    }
+
                     // Move this object to where the raycast
                     // hit the Spatial Mapping mesh.
                     // Here is where you might consider adding intelligence
                     // to how the object is placed.  For example, consider
                     // placing based on the bottom of the object's
                     // collider so it sits properly on surfaces.
-                    this.transform.position = hitInfo.point;
+                    gameObject.transform.position = hitInfo.point;
```

Comment:
Good catch, we should be checking to make we should either update the parent, or the object this script is attached to.